### PR TITLE
Per issue #5

### PR DESCRIPTION
**Pull request by tresf**

### CODE REVIEW

**Comment by akberenz** [comment]
Since this is suppose to fix an issue for a particular mac, can you add the address to the test data in the Stepper file?

**Comment by tresf** [comment]
> Since this is suppose to fix an issue for a particular mac, can you add the address to the test data in the Stepper file?

@bberenz I was thinking the same thing but first I wanted your opinion on the fix.  e.g. should this be fixed on the Java side instead?

File: sift.js [hunk 1]
```diff
@@ -44,7 +44,8 @@ var Sifter = (function() {
         nonBurnedAddressMasks: [], //populated by constructor calls
 
         burned: function(address) {
-            if (!address) { return true; }
+            if (address === 'UNKNOWN') { return true; }
```
Comment (akberenz) [inline sift.js:47]: Our default changed from empty, this keeps it as burned in

File: sift.js [hunk 2]
```diff
@@ -44,7 +44,8 @@ var Sifter = (function() {
...
+            if (!address) { return false; }
```
Comment (akberenz) [inline sift.js:48]: We explicitly want empty addresses as non-burned now

File: sift.js [hunk 3]
```diff
@@ -59,7 +60,7 @@ var Sifter = (function() {
         },
 
         plain: function(address) {
-            return address ? address.replace(/[^A-Fa-f0-9]/g, "").toUpperCase() : address;
+            return (address || "").replace(/[^A-Fa-f0-9]/g, "").toUpperCase();
```
Comment (akberenz) [inline sift.js:63]: This way we will never return `null`

File: sift.js [hunk 4]
```diff
@@ -244,9 +247,7 @@ var Sifter = (function() {
                 },
 
                 addresses: function(list, param, value) {
-                    console.log(param, value);
                     for(var i = list.length - 1; i >= 0; i--) {
-                        console.log(list[i], internal.match.addressScheme(list[i].mac));
```
Comment (akberenz) [inline sift.js:249]: These were left over from earlier development

File: tests/stepper.js [hunk 5]
```diff
@@ -177,13 +177,14 @@ var Stepper = {
...
                 },
-                { mac: '00-16-3E-00-FF-FF', details: { burnedIn: false, vmGuest: true } },
-                { mac: '02C01DC0FFEE', details: { burnedIn: true, vmGuest: false } },
-                { mac: '4A:CC:E5:52:AB:ED', details: { burnedIn: true, vmGuest: false } },
-                { mac: '00-00-00-00-00-00-00-E0', details: { burnedIn: false, vmGuest: false } }
+                { mac: '00-16-3E-00-FF-FF', primary: false, details: { burnedIn: false, vmGuest: true } },
+                { mac: '02C01DC0FFEE', primary: false, details: { burnedIn: true, vmGuest: false } },
+                { mac: '4A:CC:E5:52:AB:ED', primary: false, details: { burnedIn: true, vmGuest: false } },
+                { mac: '00-00-00-00-00-00-00-E0', primary: false, details: { burnedIn: false, vmGuest: false } },
+                { mac: null, primary: false, details: { burnedIn: false, vmGuest: false } }
```
Comment (akberenz) [inline tests/stepper.js:187]: Add `null` test case, plus add `primary` as it's the new magic proprty for address filtering.

File: sift.js [hunk 6]
```diff
@@ -214,7 +217,7 @@ var Sifter = (function() {
 
                 for(var i = 1; i < internal.networkAdapters.length; i++) {
                     var adapter = internal.networkAdapters[i];
-                    if (adapter.mac === plainAddress || (plainAddress && plainAddress.indexOf(adapter.mac) == 0)) {
+                    if (adapter.mac === plainAddress || (adapter.mac && plainAddress.indexOf(adapter.mac) == 0)) {
```
Comment (akberenz) [inline sift.js:220]: `plainAddress` is never null now, but we don't want to match our new `null` address on the first index of every adapter mac

**Comment by akberenz** [comment]
The review feature didn't do quite what I thought it would do, but there's my suggested revisions to this.
I don't think handling this java would be a good solution, as this case may come in from another source than just QZ.

**Comment by tresf** [comment]
> The review feature didn't do quite what I thought it would do, but there's my suggested revisions to this.

Thanks.  Are you OK pulling down and editing the PR directly?

> I don't think handling this java would be a good solution, as this case may come in from another source than just QZ.

Makes sense.

**Comment by akberenz** [comment]
> Are you OK pulling down and editing the PR directly?

I did, this is all from a commit (769d1d2df3941de640d3610b319859095f5802d9), I just didn't realize it would throw all the comments I made here as well.

**Pull request merged**